src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -142,22 +157,40 @@ protected function validateSameSiteConfiguration(array $configuration): bool
     }
 
     /**
-     * {@inheritdoc}
+     * @return OptionsResolver
      */
-    public function validateConfiguration(array $configuration)
+    protected function buildResolver(): OptionsResolver
     {
-        return isset(
-            $configuration['enable_ssl'],
-            $configuration['enable_ssl_everywhere'],
-            $configuration['enable_token'],
-            $configuration['allow_html_iframes'],
-            $configuration['use_htmlpurifier'],
-            $configuration['price_round_mode'],
-            $configuration['price_round_type'],
-            $configuration['display_suppliers'],
-            $configuration['display_manufacturers'],
-            $configuration['display_best_sellers'],
-            $configuration['multishop_feature_active']
-        );
+        $resolver = (new OptionsResolver())
+            ->setDefined(
+                [
+                    'enable_ssl',
+                    'enable_ssl_everywhere',
+                    'enable_token',
+                    'allow_html_iframes',
+                    'use_htmlpurifier',
+                    'price_round_mode',
+                    'price_round_type',
+                    'display_suppliers',
+                    'display_manufacturers',
+                    'display_best_sellers',
+                    'multishop_feature_active',
+                    'shop_activity',
+                ]
+            )
+            ->setAllowedTypes('enable_ssl', 'bool')
+            ->setAllowedTypes('enable_ssl_everywhere', 'bool')
+            ->setAllowedTypes('enable_token', 'bool')
+            ->setAllowedTypes('allow_html_iframes', 'bool')
+            ->setAllowedTypes('use_htmlpurifier', 'bool')
+            ->setAllowedTypes('price_round_mode', 'integer')
+            ->setAllowedTypes('price_round_type', 'integer')
+            ->setAllowedTypes('display_suppliers', 'bool')
+            ->setAllowedTypes('display_manufacturers', 'bool')
+            ->setAllowedTypes('display_best_sellers', 'bool')
+            ->setAllowedTypes('multishop_feature_active', 'bool')
+            ->setAllowedTypes('shop_activity', 'integer');
```

Comment:
What do you think about adding some constraint for integer configurations according `PrestaShopBundle\Form\Admin\Configure\ShopParameters\General\PreferencesType` ?

```suggestion
            ->setAllowedTypes('price_round_mode', 'integer')
            ->setAllowedValues('price_round_mode', [
            	$this->configuration->get('PS_ROUND_HALF_UP'),
	            $this->configuration->get('PS_ROUND_HALF_DOWN'),
                $this->configuration->get('PS_ROUND_HALF_EVEN'),
                $this->configuration->get('PS_ROUND_HALF_ODD'),
                $this->configuration->get('PS_ROUND_UP'),
                $this->configuration->get('PS_ROUND_DOWN'),
        	])
            ->setAllowedTypes('price_round_type', 'integer')
            ->setAllowedValues('price_round_type', [
            	Order::ROUND_ITEM,
            	Order::ROUND_LINE,
            	Order::ROUND_TOTAL,
        	])
            ->setAllowedTypes('display_suppliers', 'bool')
            ->setAllowedTypes('display_manufacturers', 'bool')
            ->setAllowedTypes('display_best_sellers', 'bool')
            ->setAllowedTypes('multishop_feature_active', 'bool')
            ->setAllowedTypes('shop_activity', 'integer')
            ->setAllowedValues('shop_activity', range(1, 20));
```